utility/app.go

```diff
@@ -0,0 +1,431 @@
+package utility
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	utilTypes "github.com/pokt-network/pocket/utility/types"
+)
+
+func (u *UtilityContext) HandleMessageStakeApp(message *utilTypes.MessageStakeApp) types.Error {
+	publicKey, er := crypto.NewPublicKeyFromBytes(message.PublicKey)
+	if er != nil {
+		return types.ErrNewPublicKeyFromBytes(er)
+	}
+	// ensure above minimum stake
+	minStake, err := u.GetAppMinimumStake()
+	if err != nil {
```

Comment:
The amount of `err != nil` in this function is painful. Go should improve on this tbh, but one way to reduce this is:

```
minStake, err := u.GetAppMinimumStake()
check(err) // panic on non-nil
```

and at the top of the function:
```
defer func() {
    // logic to recover from panic and return an error instead.
}
```

Reply:
Adding a protocol hour's topic